src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -372,6 +372,22 @@ protected virtual ComponentState CreateComponentState(int componentId, IComponen
     /// has completed.
     /// </returns>
     public virtual Task DispatchEventAsync(ulong eventHandlerId, EventFieldInfo? fieldInfo, EventArgs eventArgs)
+    {
+        return DispatchEventAsync(eventHandlerId, fieldInfo, eventArgs, quiesce: false);
+    }
+
+    /// <summary>
+    /// Notifies the renderer that an event has occurred.
+    /// </summary>
+    /// <param name="eventHandlerId">The <see cref="RenderTreeFrame.AttributeEventHandlerId"/> value from the original event attribute.</param>
+    /// <param name="eventArgs">Arguments to be passed to the event handler.</param>
+    /// <param name="fieldInfo">Information that the renderer can use to update the state of the existing render tree to match the UI.</param>
+    /// <param name="quiesce">Whether to wait for quiescence or not.</param>
+    /// <returns>
+    /// A <see cref="Task"/> which will complete once all asynchronous processing related to the event
+    /// has completed.
+    /// </returns>
+    public virtual Task DispatchEventAsync(ulong eventHandlerId, EventFieldInfo? fieldInfo, EventArgs eventArgs, bool quiesce)
```

Comment:
We need a new overload to track quiesce when we are dispatching an event to a named event handler. Alternatively, we can make quiescence a feature the renderer offers, and in that way, we don't have to be adding it to every API.

Reply:
It's surprising to me that we need to return any quiescence-related `Task` from here. I would have expected that when `Endpoints` dispatches the event, that simply extends the existing `QuiescenceTask` that it was going to wait for anyway.

Perhaps that doesn't work out for some reason that is hard to understand. It would be great to clean this up but I would certainly understand if you don't want to block this PR on it. So I suggest:

 * If you know clearly why it's *not* right for the existing `QuiescenceTask` to be extended by this dispatch, and don't think we should change it to do that, could you say why?
 * Or if you think it would be reasonable for the existing `QuiescenceTask` to be extended by this dispatch, could you file an issue saying we should do that? I'd be happy to dig into what's going on and find out if I can do that.

Reply:
@SteveSandersonMS Everything is triggered from
https://github.com/dotnet/aspnetcore/blob/8968058c9e5fdfdd1242426a03dc80609997edab/src/Components/Components/src/RenderTree/Renderer.cs#L229-L238

But when we wait for quiescence, we reset it after we are done (note the finally block):
https://github.com/dotnet/aspnetcore/blob/8968058c9e5fdfdd1242426a03dc80609997edab/src/Components/Components/src/RenderTree/Renderer.cs#L293-L312